examples/blas/compile_time_dispatching/level3/gemm_usm_mklcpu_cublas.cpp

```diff
@@ -0,0 +1,276 @@
+    std::fill(result_gpu.begin(), result_gpu.end(), 0);
+
+    rand_matrix(A, transA, m, k, ldA);
+    rand_matrix(B, transB, k, n, ldB);
+    rand_matrix(C, oneapi::mkl::transpose::nontrans, m, n, ldC);
+
+    //
+    // Preparation on CPU
+    //
+    sycl::queue cpu_queue(cpu_dev, cpu_exception_handler);
+    sycl::event cpu_gemm_done;
+    sycl::context cpu_cxt = cpu_queue.get_context();
+
+    // allocate on CPU device and copy data from host to SYCL CPU device
+    float *cpu_A = sycl::malloc_device<float>(sizea * sizeof(float), cpu_queue);
+    float *cpu_B = sycl::malloc_device<float>(sizeb * sizeof(float), cpu_queue);
+    float *cpu_C = sycl::malloc_device<float>(sizec * sizeof(float), cpu_queue);
+    if (!cpu_A || !cpu_B || !cpu_C) {
+        throw std::runtime_error("Failed to allocate USM memory.");
+    }
+    cpu_queue.memcpy(cpu_A, A.data(), sizea * sizeof(float)).wait();
+    cpu_queue.memcpy(cpu_B, B.data(), sizeb * sizeof(float)).wait();
+    cpu_queue.memcpy(cpu_C, C.data(), sizec * sizeof(float)).wait();
+
+    //
+    // Preparation on GPU
+    //
+    sycl::queue gpu_queue(gpu_dev, gpu_exception_handler);
+    sycl::event gpu_gemm_done;
+    sycl::context gpu_cxt = gpu_queue.get_context();
+
+    // allocate on GPU device and copy data from host to SYCL GPU device
+    float *gpu_A = sycl::malloc_device<float>(sizea * sizeof(float), gpu_queue);
+    float *gpu_B = sycl::malloc_device<float>(sizeb * sizeof(float), gpu_queue);
+    float *gpu_C = sycl::malloc_device<float>(sizec * sizeof(float), gpu_queue);
+    if (!gpu_A || !gpu_B || !gpu_C) {
+        throw std::runtime_error("Failed to allocate USM memory.");
+    }
+    gpu_queue.memcpy(gpu_A, A.data(), sizea * sizeof(float)).wait();
+    gpu_queue.memcpy(gpu_B, B.data(), sizeb * sizeof(float)).wait();
+    gpu_queue.memcpy(gpu_C, C.data(), sizec * sizeof(float)).wait();
+
+    //
+    // Execute Gemm on CPU and GPU device
+    //
+    // add oneapi::mkl::blas::gemm to execution queue
+    cpu_gemm_done = oneapi::mkl::blas::column_major::gemm(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::mklcpu>{ cpu_queue }, transA, transB, m,
+        n, k, alpha, cpu_A, ldA, cpu_B, ldB, beta, cpu_C, ldC);
+    gpu_gemm_done = oneapi::mkl::blas::column_major::gemm(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cublas>{ gpu_queue }, transA, transB, m,
+        n, k, alpha, gpu_A, ldA, gpu_B, ldB, beta, gpu_C, ldC);
+
+    // Wait until calculations are done
+    cpu_gemm_done.wait_and_throw();
+    gpu_gemm_done.wait_and_throw();
+
+    //
+    // Post Processing
+    //
+    // copy data from CPU back to host
+    cpu_queue.memcpy(result_cpu.data(), cpu_C, sizec * sizeof(float)).wait_and_throw();
+
+    // copy data from GPU back to host
+    gpu_queue.memcpy(result_gpu.data(), gpu_C, sizec * sizeof(float)).wait_and_throw();
+
+    // compare
+    int ret = check_equal_matrix(result_cpu.data(), result_gpu.data(), m, n, ldC);
```

Comment:
Do we use this compare result to check PASS/FAIL?

Reply:
No, compare result is only printed in std::cout (or std::cerr if failed). Ctest will still pass because we do not exit with non-zero error code

Reply:
I see. I find it misleading though. Comparing the results indicate that, the test should fail if the results are not match. Can we print the matrices like RT example instead of comparing the results?

Reply:
Changed to printing result matrices instead of comparing them.